openmc/deplete/stepresult.py

```diff
@@ -318,10 +318,11 @@ def _write_hdf5_metadata(self, handle):
                               chunks=(1, 1, n_mats, n_nuc_number),
                               dtype='float64')
 
-        handle.create_dataset("reaction rates", (1, n_stages, n_mats, n_nuc_rxn, n_rxn),
-                              maxshape=(None, n_stages, n_mats, n_nuc_rxn, n_rxn),
-                              chunks=(1, 1, n_mats, n_nuc_rxn, n_rxn),
-                              dtype='float64')
+        if n_nuc_rxn > 0 and n_rxn > 0:
+            handle.create_dataset("reaction rates", (1, n_stages, n_mats, n_nuc_rxn, n_rxn),
+                                maxshape=(None, n_stages, n_mats, n_nuc_rxn, n_rxn),
+                                chunks=(1, 1, n_mats, n_nuc_rxn, n_rxn),
+                                dtype='float64')
```

Comment:
It's a little stronger than just not writing an empty dataset -- if no reactions exist on the `MicroXS` object, this will actually fail because `h5py` will refuse to create a dataset where one of the dimensions has size 0.